pkg/controller/volume/attach_detach_controller.go

```diff
@@ -90,7 +136,95 @@ func NewAttachDetachController(
 		DeleteFunc: adc.nodeDelete,
 	})
 
-	return adc
+	if err := adc.volumePluginMgr.InitPlugins(plugins, adc); err != nil {
+		return nil, fmt.Errorf("Could not initialize volume plugins for Attach/Detach Controller: %+v", err)
+	}
+
+	adc.desiredStateOfWorld = cache.NewDesiredStateOfWorld(&adc.volumePluginMgr)
+	adc.actualStateOfWorld = cache.NewActualStateOfWorld(&adc.volumePluginMgr)
```

Comment:
I am still reading the PR but is it possible to use the same data structure for actual and desired state of the world ? if so wouldn't that make the reconcile easier ?